src/lex/replace.rs

```diff
@@ -312,3 +325,29 @@ fn replace_function(
         .chain(replacements.into_iter().map(|t| Ok(location.with(t))))
         .collect()
 }
+
+fn stringify(args: Vec<Token>) -> Token {
+    let ret: String = args
+        .into_iter()
+        .map(|arg| {
+            match arg {
+                Token::Whitespace(_) => String::from(" "), // Single space in replacement
+                Token::Literal(Literal::Str(s)) => {
+                    // TODO can we unwrap safely?
+                    let new_s = std::str::from_utf8(&s)
+                        .unwrap()
+                        .trim_end_matches('\u{0}')  // remove null terminator
+                        .escape_default()  // Escape whitespace, etc
+                        .flat_map(char::escape_default); // Escape a second time
+                    "\\\"" // Wrap with escaped quotation marks
+                        .chars()
+                        .chain(new_s)
+                        .chain("\\\"".chars())
+                        .collect::<String>()
+                }
+                other => other.to_string(),
+            }
+        })
+        .collect();
+    Token::Literal(Literal::Str(ret.into()))
```

Comment:
This code looks right, but the behavior is wrong:
```c
#define str(a) # a
str(b)
// output
 b 
```
For comparison, clang outputs `"b"`.

Reply:
Ah, I found the issue - this should ignore whitespace tokens. For comparison, this works fine:
```c
#define f(a) #a
f(a)   
// output
"a"
```